src/shared/components/plots/PlotsTabUtils.tsx

```diff
@@ -1146,101 +1146,137 @@ export function makeAxisDataPromise_Molecular_MakeMutationData(
+                hugoGeneSymbol
+            ),
+            isSampleProfiled => isSampleProfiled === true
+        );
+
+        switch (mutationCountBy) {
+            case MutationCountBy.MutationType:
+                if (!sampleMutations) {
+                    value = isProfiled
+                        ? MUT_PROFILE_COUNT_NOT_MUTATED
+                        : MUT_PROFILE_COUNT_NOT_PROFILED;
+                } else {
+                    const types = _.uniq(
                         sampleMutations.map(
                             m =>
                                 mutationTypeToDisplayName[
                                     getOncoprintMutationType(m)
                                 ]
                         )
                     );
-                    // if more than one type, its "Multiple"
-                    if (sampleMutTypes.length > 1) {
-                        plotValue = MUT_PROFILE_COUNT_MULTIPLE;
-                    } else {
-                        plotValue = sampleMutTypes;
-                    }
-                    break;
-                case MutationCountBy.DriverVsVUS:
-                    if (_.some(sampleMutations, m => m.putativeDriver)) {
-                        plotValue = MUT_PROFILE_COUNT_DRIVER;
-                    } else {
-                        plotValue = MUT_PROFILE_COUNT_VUS;
-                    }
-                    break;
-                case MutationCountBy.MutatedVsWildType:
-                default:
-                    plotValue = MUT_PROFILE_COUNT_MUTATED;
-                    break;
-            }
+                    value =
+                        types.length > 1 ? MUT_PROFILE_COUNT_MULTIPLE : types;
+                }
+                break;
+
+            case MutationCountBy.DriverVsVUS:
+                if (!sampleMutations) {
+                    value = isProfiled
+                        ? MUT_PROFILE_COUNT_NOT_MUTATED
+                        : MUT_PROFILE_COUNT_NOT_PROFILED;
+                } else {
+                    value = _.some(sampleMutations, m => m.putativeDriver)
+                        ? MUT_PROFILE_COUNT_DRIVER
+                        : MUT_PROFILE_COUNT_VUS;
+                }
+                break;
+
+            case MutationCountBy.VAF:
+                if (!sampleMutations) {
+                    value = isProfiled ? 0 : NaN;
+                } else {
+                    const vafs = sampleMutations
+                        .map(m => {
+                            if (
+                                Number.isInteger(m.tumorAltCount) &&
+                                Number.isInteger(m.tumorRefCount)
+                            ) {
+                                const total = m.tumorAltCount + m.tumorRefCount;
+                                return total > 0
+                                    ? m.tumorAltCount / total
+                                    : null;
+                            }
+                            return null;
```

Comment:
Let's reuse an existing utility function for this if possible.

https://github.com/cBioPortal/cbioportal-frontend/blob/2894692dc0ae1e02b7985aa2fe49fb44f9c7631d/src/shared/lib/MutationUtils.tsx#L382-L399